x/committee/client/common/query.go

```diff
@@ -73,10 +73,10 @@ func QueryProposalByID(cliCtx client.Context, cdc *codec.LegacyAmino, queryRoute
 	res, height, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", queryRoute, types.QueryProposal), bz)
 
 	if err == nil {
-		var proposal *types.Proposal
+		var proposal types.Proposal
 		cdc.MustUnmarshalJSON(res, &proposal)
 
-		return proposal, height, nil
+		return &proposal, height, nil
```

Comment:
@DracoLi So in this case the value of `proposal` is still undefined(?) but the **pointer** has been updated to by `MustUnmarshalJSON` to have data?

Reply:
I believe we were passing the address of a pointer to unmarshall before. That didn't work well with proto's unpack interface, which resulted in the content (which requires unpacking) to not be processed, thus resulting in a null value.